module/zfs/zfs_ioctl.c

```diff
@@ -5821,6 +5823,54 @@ zfs_ioc_send_space(const char *snapname, nvlist_t *innvl, nvlist_t *outnvl)
 	return (error);
 }
 
+/*
+ * Sync the currently open TXG to disk for the specified pool.
+ * This is somewhat similar to 'zfs_sync()'.
+ * For all cases that do not result in error this ioctl will wait for
+ * the currently open TXG to commit before returning back to the caller.
+ *
+ * innvl is unused
+ *
+ * onvl is unused
+ */
+/* ARGSUSED */
+static int
+zfs_ioc_pool_sync(const char *pool, nvlist_t *innvl, nvlist_t *onvl)
+{
+	int err;
+	spa_t *spa;
+
+	ASSERT3P(innvl, ==, NULL);
+
+	if ((err = spa_open(pool, &spa, FTAG)) != 0)
+		return (err);
+
+	if (spa->spa_failmode == ZIO_FAILURE_MODE_PANIC) {
+		VERIFY3S(spa_writeable(spa), ==, B_TRUE);
+		VERIFY3S(spa_suspended(spa), ==, B_FALSE);
+		VERIFY3S(spa_state(spa), ==, POOL_STATE_ACTIVE);
+	} else if (spa->spa_failmode == ZIO_FAILURE_MODE_WAIT) {
+		if (spa_writeable(spa) == B_FALSE ||
+		    spa_suspended(spa) == B_TRUE ||
+		    spa_state(spa) != POOL_STATE_ACTIVE) {
+			spa_close(spa, FTAG);
+			return (EAGAIN);
+		}
+	} else if (spa->spa_failmode != ZIO_FAILURE_MODE_CONTINUE) {
+		spa_close(spa, FTAG);
+		return (EINVAL);
+	}
+
+	/* force a TXG sync if there isn't one happening */
+	txg_kick(spa_get_dsl(spa));
```

Comment:
My understanind was the without txg_kick() we may end up waiting the full txg timeout. I'll double check this